sandbox-web/src/main/java/com/exadel/sandbox/team5/controller/DiscountRestController.java

```diff
@@ -49,5 +56,19 @@ public void delete(@PathVariable Long id) {
     public List<Review> getReviewsByDiscount(@PathVariable Long discountId) {
         return reviewService.getReviewsByDiscount(discountId);
     }
+
+    @ApiOperation("Generating QR code with text \"Exadel employee. Special discount\"")
+    @PostMapping(value = "/qrcode", produces = MediaType.IMAGE_PNG_VALUE)
+    @ResponseStatus(HttpStatus.OK)
+    public ResponseEntity<byte[]> generateQRCode() {
+        try (var baos = new ByteArrayOutputStream()) {
+            var image = QRCodeGenerator.generateQRCodeImage("Exadel employee. Special discount");
```

Comment:
We can return `byte[]` in the service method. Also it will be great to have at least one unit test for this service method

Reply:
Moved in service layer,
I tried create a test.